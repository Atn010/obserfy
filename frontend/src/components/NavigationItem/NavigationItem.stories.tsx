import React, { FC } from "react"
import NavigationItem from "./NavigationItem"
import { ReactComponent as HomeIcon } from "../../icons/home.svg"
import SideBar from "../SideBar/SideBar"

export default {
  title: "Basic|Navigation/NavigationItem",
  component: NavigationItem,
  parameters: {
    componentSubtitle: "Component that acts as a navigation link"
  }
}

export const Basic: FC = () => (
  <SideBar>
    <NavigationItem mt={3} icon={HomeIcon} to="/somewhere" text="home" />
  </SideBar>
)
