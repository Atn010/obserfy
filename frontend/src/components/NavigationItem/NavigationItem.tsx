import React, { FC, useState } from "react"
import Typography from "../Typography/Typography"
import Icon from "../Icon/Icon"
import Flex, { FlexProps } from "../Flex/Flex"
import { Link } from "gatsby-plugin-intl3"

interface Props extends FlexProps {
  text: string
  icon?: FC
  to: string
}

/** This component represents a navigation target. currently only used
 * on the sidebar. It uses Gatsby's link to render the ```<a></a>``` tag for creating
 * a link to other page on the app.
 * */
export const NavigationItem: FC<Props> = ({
  text,
  icon,
  to,
  ...props
}) => {
  const [isActive, setIsActive] = useState(false)
  return (
    <Link
      to={to}
      getProps={({ isCurrent, isPartiallyCurrent }) => {
        setIsActive(to === "/" ? isCurrent : isPartiallyCurrent)
        return {} // for satisfying types request, no idea why tho....
      }}
    >
      <Flex
        alignItems="center"
        m={2}
        my={0}
        pl={2}
        backgroundColor={isActive ? "primaryLighter" : "none"}
        sx={{
          borderRadius: "default",
          textDecoration: "none",
          "&:hover": {
            backgroundColor: isActive ? "primaryLighter" : "primaryLightest"
          }
        }}
        {...props}
      >
        <Icon
          alt={`${text} icon`}
          ml={0}
          my={2}
          as={icon}
          sx={{ fill: isActive ? "primaryDark" : "icon" }}
        />
        <Typography.Body
          fontWeight={isActive ? "bold" : "normal"}
          color={isActive ? "primaryDark" : "text"}
          m={0}
        >
          {text}
        </Typography.Body>
      </Flex>
    </Link>
  )
}

export default NavigationItem
