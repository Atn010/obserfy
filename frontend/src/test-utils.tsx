/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
