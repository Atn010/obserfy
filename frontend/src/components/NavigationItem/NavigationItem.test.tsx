import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./NavigationItem.stories"

describe("NavigationItem", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
