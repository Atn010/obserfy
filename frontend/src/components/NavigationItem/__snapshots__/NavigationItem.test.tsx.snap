// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem should render correctly 1`] = `
.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-7 {
    font-size: 20px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 360px;
  min-width: 230px;
  position: fixed;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(0.0,0.0,0.2,1);
  -webkit-transition: transform 250ms cubic-bezier(0.0,0.0,0.2,1);
  transition: transform 250ms cubic-bezier(0.0,0.0,0.2,1);
  top: 0;
  left: 0;
  z-index: 1250000;
  border-right-style: solid;
  border-right-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  border-right-width: 1px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
  width: 82%;
  height: calc(100% + 57px);
  padding-bottom: 65px;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:40em) {
  .emotion-3 {
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    box-shadow: none;
    z-index: 0;
  }
}

@media screen and (min-width:40em) {
  .emotion-3 {
    width: 230px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px;
  margin-top: 16px;
  margin-bottom: 0;
  padding-left: 8px;
  background-color: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2:hover {
  background-color: var(--theme-ui-colors-primaryLightest,rgba(98,151,73,0.05));
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 16px;
  height: 20px;
  width: 20px;
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-size: 16px;
  font-weight: normal;
  margin: 0;
}

@media screen and (min-width:40em) {
  .emotion-1 {
    font-size: 14px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  display: inherit;
  pointer-events: inherit;
  position: fixed;
  top: 0;
  left: 0;
  min-height: 100vh;
  z-index: 1000001;
  -webkit-transition: background 250ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background 250ms cubic-bezier(0.0,0.0,0.2,1);
  background-color: var(--theme-ui-colors-overlay,rgba(0,0,0,.6));
  width: 100%;
  height: 100%;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    display: none;
  }
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  right: 0;
  z-index: 1500000;
  height: 57px;
  width: 18vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width:40em) {
  .emotion-6 {
    display: none;
  }
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-onOverlay,#fff);
  width: 36px;
  height: 36px;
}

<div>
  <div
    class="emotion-7"
  >
    <nav
      class="emotion-3"
      data-cy="sidebar"
    >
      <a
        href="/somewhere"
      >
        <div
          class="emotion-2"
        >
          <div
            alt="home icon"
            class="emotion-0"
          />
          <p
            class="emotion-1"
          >
            home
          </p>
        </div>
      </a>
    </nav>
    <div
      class="emotion-4"
      data-cy="outsideSidebar"
    />
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      />
    </div>
  </div>
</div>
`;
